Session/Settings/AppIconViewModel.swift

```diff
@@ -137,6 +137,12 @@ class AppIconViewModel: SessionTableViewModel, NavigatableStateHolder, Observabl
     lazy var observation: TargetObservation = ObservationBuilderOld
         .subject(selectedOptionsSubject)
         .mapWithPrevious { [weak self, dependencies] previous, current -> [SectionModel] in
+            
+            if let currentIcon = current {
+                // Save latest app icon disguise selected
+                dependencies[defaults: .standard, key: .lastSelectedAppIconDisguise] = currentIcon
+            }
+            
```

Comment:
This is an odd place to update the value, please move it to `updateAppIcon` instead (just above `selectedOptionsSubject`)